src/archivist/fetcher/fullDomFetcher.js

```diff
@@ -86,7 +86,21 @@ export async function launchHeadlessBrowser() {
     return browser;
   }
 
-  browser = await puppeteer.launch({ headless: true });
+  const options = {
+    args: [],
+    headless: true,
+  };
+  if (process.env.http_proxy) {
+    options.args = [].concat(options.args, `--proxy-server=${process.env.http_proxy}`);
```

Comment:
I thought the issue was Puppeteer not supporting proxy authentication directly via the proxy URL (as you mentioned [here](https://github.com/OpenTermsArchive/engine/pull/1144#issuecomment-2901572993)). The expected fix, as I understood it, was to extract the username and password from the URL and pass them to `await page.authenticate({ username, password })`. However, I didn’t see this implemented, did I misunderstand something?

Reply:
What about `process.env.HTTPS_PROXY`?

Reply:
This logic is ok for me.